src/components/LiquidityMining/Create/Steps/PreviewAndCreate/SimulateStaking/index.tsx

```diff
@@ -0,0 +1,229 @@
+  padding: 8px 12px;
+  align-self: center;
+`
+
+const SimulateOption = styled.div<{ isActive: boolean }>`
+  font-size: 11px;
+  font-weight: 600;
+  letter-spacing: 0.08em;
+  color: ${props => (props.isActive ? props.theme.text2 : props.theme.lightPurple2)};
+  ${props => props.isActive && 'text-decoration: underline; text-underline-offset: 7px;'};
+`
+const StyledNumericalInput = styled(NumericalInput)<{ value: string }>`
+  color: ${props => props.theme.text2};
+  width: ${({ value }) => (value && value.length > 24 ? '250' : value ? 17 + value?.length * 8 : 20)}px;
+  max-height: 38px;
+  font-weight: 600;
+  font-size: 16px;
+  line-height: 16px;
+  text-transform: uppercase;
+  ::placeholder {
+    color: ${props => props.theme.text2};
+  }
+`
+
+const DOLLAR_AMOUNT_MAX_SIMULATION = 10000000
+
+interface RewardSummaryProps {
+  tokenOrPair: Token | Pair | null
+  stakingCap: TokenAmount | null
+  nativeCurrencyUSDPrice: Price
+  setSimulatedStakedAmount: (value: string) => void
+  setSimulatedPrice: (value: any) => void
+  simulatedPrice: any
+  loading: boolean
+}
+enum SimulateOptions {
+  AMOUNT = 'Amount',
+  PRICE = 'Price',
+}
+
+export default function SimulateStaking({
+  tokenOrPair,
+  stakingCap,
+  setSimulatedStakedAmount,
+  setSimulatedPrice,
+  simulatedPrice,
+  nativeCurrencyUSDPrice,
+  loading,
+}: RewardSummaryProps) {
+  const { loading: loadingNativeTokenPrice, derivedNativeCurrency: nativeTokenPrice } = useTokenOrPairNativeCurrency(
+    tokenOrPair ? tokenOrPair : undefined
+  )
+
+  const handleLocalStakingCapChange = useCallback(
+    rawValue => {
+      setSimulatedPrice(rawValue)
+    },
+    [setSimulatedPrice]
+  )
+  useEffect(() => {
+    if (nativeTokenPrice && nativeCurrencyUSDPrice)
+      setSimulatedPrice(nativeTokenPrice.multiply(nativeCurrencyUSDPrice).toFixed(2))
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [loadingNativeTokenPrice, nativeTokenPrice])
+
+  const [simulateOption, setSimulateOption] = useState<SimulateOptions>(SimulateOptions.AMOUNT)
+  const [showUSDValue, setShowUSDValue] = useState(true)
+  const [simulatedValuePercentage, setSimulatedValuePercentage] = useState(10)
+  const liquidityPercentChangeCallback = useCallback((value: number) => {
+    setSimulatedValuePercentage(value)
+  }, [])
+  const [innerLiquidityPercentage, setInnerLiquidityPercentage] = useDebouncedChangeHandler(
+    simulatedValuePercentage,
+    liquidityPercentChangeCallback,
+    10
+  )
+  const maxStakedSimulatedAmount = useMemo(() => {
+    const simulatedPriceAdjusted = simulatedPrice || '0'
+
+    const base = stakingCap
+      ? parseFloat(stakingCap.multiply(simulatedPriceAdjusted).toSignificant(22))
+      : DOLLAR_AMOUNT_MAX_SIMULATION
+
+    const baseInUsd = parseFloat(simulatedPriceAdjusted)
+
+    const baseValue = showUSDValue ? base : base / baseInUsd
+
+    const baseCurrency = tokenOrPair instanceof Token ? tokenOrPair : tokenOrPair?.liquidityToken
```

Comment:
```suggestion
    const isToken = tokenOrPair instanceof Token
    const baseCurrency = isToken ? tokenOrPair : tokenOrPair?.liquidityToken
```

There are other places we can use this constant too.